deeplab/README.md

```diff
@@ -0,0 +1,202 @@
+
+This package contains a standalone implementation of the DeepLab inference pipeline, as well as a [demo](./demo), for running semantic segmentation using TensorFlow.js.
+
+![DeepLab Demo](./docs/deeplab-demo.gif)
+
+## Usage
+
+In the first step of semantic segmentation, an image is fed through a pre-trained model [based](https://github.com/tensorflow/models/blob/master/research/deeplab/g3doc/model_zoo.md) on MobileNet-v2. Three types of pre-trained weights are available, trained on [Pascal](http://host.robots.ox.ac.uk/pascal/VOC/voc2012/index.html), [Cityscapes](https://www.cityscapes-dataset.com) and [ADE20K](https://groups.csail.mit.edu/vision/datasets/ADE20K/) datasets.
+
+To get started, pick the model name from `pascal`, `cityscapes` and `ade20k`, and decide whether you want your model quantized to 1 or 2 bytes (set the `quantizationBytes` option to 4 if you want to disable quantization). Then, initialize the model as follows:
+
+```typescript
+import {load} from '@tensorflow-models/deeplab';
+const modelName = 'pascal';   // set to your preferred model, out of `pascal`, `cityscapes` and `ade20k`
+const quantizationBytes = 2; // either 1, 2 or 4
+const modelPromise = load({base: modelName, quantizationBytes});
+```
+
+By default, calling `load` initalizes the PASCAL variant of the model quantized to 2 bytes.
+
+If you would rather load custom weights, you can pass the URL in the config instead:
+
+```typescript
+import {load} from '@tensorflow-models/deeplab';
+const modelName = 'pascal';   // set to your preferred model, out of `pascal`, `cityscapes` and `ade20k`
+const quantizationBytes = 2; // either 1, 2 or 4
+const url = 'https://storage.googleapis.com/gsoc-tfjs/models/deeplab/quantized/1/pascal/model.json';
+const modelPromise = load({modelUrl: url});
+```
+
+This will initialize the model and return the promise of a `SemanticSegmentation` object.
+
+You can still pass the `base` attribute set to either `pascal`,`pascal`, `cityscapes` and `ade20k` in the argument to use the corresponding colormap and labelling scheme. Otherwise, you would have to provide those yourself.
+
+If you require more careful control over the initialization and behavior of the model (e.g. you want to use your own labelling scheme and colormap), use the `SemanticSegmentation` object on its own, passing a pre-loaded `GraphModel` in the constructor:
+
+```typescript
+import * as tfconv from '@tensorflow/tfjs-converter';
+import {SemanticSegmentation, getURL} from '@tensorflow-models/deeplab';
+// use the getURL utility function to get the URL to the pre-trained weights
+const base = 'pascal';   // set to your preferred model, out of `pascal`, `cityscapes` and `ade20k`
+const quantizationBytes = 2; // either 1, 2 or 4
+const modelUrl = getURL(base, quantizationBytes);
+const rawModel = tfconv.loadGraphModel(modelUrl);
+const modelPromise = (async () => {return new SemanticSegmentation(await rawModel)();
```

Comment:
```suggestion
const model = new SemanticSegmentation(graphModel);
```

Reply:
You would need to pass `base` or give a default value otherwise `model.segment` will fail since `this.base` will be `undefined` and `getColormap` and `getLabels` will throw an error. I think there are couple of ways to move forward:

1. Either we don't give user an option to provide their own url like other tfjs-models. I think this is the best way forward and simplifies usage a lot.
2. Check all flows where code might break and fix them.

I'll say let's stick with 1st point.